File: assurapp-api/src/main/java/be/helha/assurapp/expertise/models/Claim.java
package be.helha.assurapp.expertise.models;

import be.helha.assurapp.authentication.models.User;
import be.helha.assurapp.expertise.enums.ClaimStatus;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import jakarta.persistence.Id;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "claims")
public class Claim {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    private String description;

    @Temporal(TemporalType.DATE)
    private Date date;

    @NotNull
    @Enumerated(EnumType.STRING)
    private ClaimStatus status;

    private Long expertise;

    // client
    private Long client;
    // insurer
    private Long insurer;

}

File: assurapp-web/src/app/interfaces/claim.ts
import { ClaimStatus } from "../enums/claim-status.enum";
import { Expertise } from "./expertise";
import {Subscription} from "rxjs";
import {User} from "./user";

export interface Claim {
  id: number;
  description: string;
  date: Date;
  status: ClaimStatus;
  client: User;
}
